### PR TITLE
Add documentation for docker images

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Add documentation for the following:
- How to build a docker image with NGINX+ and NGINX Agent
- How to build a docker image with NGINX OSS Mainline and NGINX Agent
- How to build a docker image with NGINX OSS Stable and NGINX Agent

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: hugo/content/docker-images.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+title: "Docker Images"
+draft: false
+description: "Learn how to build and run NGINX Agent docker images"
+weight: 800
+toc: true
+tags: [ "docs" ]
+docs: "DOCS-1230"
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:8]:
```suggestion
```
DOCS-1230 is "NGINX Agent Uninstall"

File: hugo/content/docker-images.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+categories: ["configuration"]
+doctypes: ["task"]
+---
+
+## Prerequisites
+1. Docker installation
+1. nginx-repo.crt and nginx-repo.key files (if using NGINX Plus)
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:15]: The prerequisites need to link to the appropriate documentation (ours or external) that explains how to complete the step, e.g.
```suggestion
1. Docker must be [installed and running](https://docs.docker.com/engine/install/)
1. (Optional) If you plan to use NGINX Plus, you will need the nginx-repo.crt and nginx-repo.key files. You can download them from [MyF5](https://my.f5.com).
```

File: hugo/content/docker-images.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+## Building NGINX Agent image with NGINX OSS
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:17]:
```suggestion
## Building NGINX Agent image with NGINX Open Source
```

File: hugo/content/docker-images.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+To build an image that contains the latest NGINX Agent and the latest mainline version of NGINX OSS run the following command:
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:19]:
```suggestion
To build an image that contains the latest NGINX Agent and the latest mainline version of NGINX run the following command:
```

File: hugo/content/docker-images.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+```shell
+$ cd scripts/docker/official/nginx-oss-with-nginx-agent/alpine/
+$ docker build -t nginx-agent . --no-cache -f ./Dockerfile.mainline
+```
+
+To build an image that contains the latest NGINX Agent and the latest stable version of NGINX OSS run the following command:
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:26]:
```suggestion
To build an image that contains the latest NGINX Agent and the latest stable version of NGINX run the following command:
```

File: hugo/content/docker-images.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+```shell
+$ cd scripts/docker/official/nginx-oss-with-nginx-agent/alpine/
+$ docker build -t nginx-agent . --no-cache -f ./Dockerfile.stable
+```
+
+## Building NGINX Agent image with NGINX Plus
+
+1. Log in to [MyF5 Customer Portal](https://account.f5.com/myf5) and download your `nginx-repo.crt` and `nginx-repo.key` files. For a trial of NGINX Plus, the files are provided with your trial package.
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:34]:
```suggestion
1. Log in to [MyF5 Customer Portal](https://account.f5.com/myf5) and download your `nginx-repo.crt` and `nginx-repo.key` files. This files are also provided with the NGINX Plus trial package.
```

File: hugo/content/docker-images.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+## Running NGINX Agent container
+
+Here is an example of how to run a NGINX Agent container:
+
+```console
+$ docker run --name nginx-agent -d nginx-agent
+```
+
+By default the NGINX Agent REST API is not enabled. To enable it, the NGINX Agent configuration needs to be updated to include the API configuration. Here is an example of what needs to be added to the `nginx-agent.conf`:
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:56]:
```suggestion
By default, the NGINX Agent REST API is not enabled. To enable the API, the NGINX Agent configuration file `nginx-agent.conf` needs to be updated to include the API configuration, for example:
```

File: hugo/content/docker-images.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+```yaml
+api:
+  host: 0.0.0.0
+  port: 8038
+```
+
+Once the nginx-agent.conf is updated with the API setting, here is an example of how to run the container with the updated `nginx-agent.conf` mounted and the port `8038` exposed:
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:64]:
```suggestion
Once the `nginx-agent.conf` file is updated with the API setting, following the previous example, you can run the container with the updated `nginx-agent.conf` mounted and the port `8038` exposed with the following command:
```

File: hugo/content/docker-images.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+```console
+$ docker run --name nginx-agent -d \
+  --mount type=bind,source="$(pwd)"/nginx-agent.conf,target=/etc/nginx-agent/nginx-agent.conf,readonly \
+  -p 127.0.0.1:8038:8038/tcp \
+  nginx-agent
+```
```
Comment (Jcahilltorre) [inline hugo/content/docker-images.md:71]:
```suggestion
```

```
leave a blank line at the end of markdown files.

**Review by Jcahilltorre — APPROVED** [review]
approved with suggestions

**Review by Dean-Coakley — APPROVED** [review]

**Pull request merged**